vagrant/guest/bin/m2setup.sh

```diff
@@ -351,7 +351,7 @@ code=
 mysql -e "use $DB_NAME" 2> /dev/null || code="$?"
 if [[ $code ]]; then
     echo "==> Creating $DB_NAME database"
-    mysql -e "create database $DB_NAME"
+    mysql -e "create database \`$DB_NAME\`"
```

Comment:
@rtull Ok, that makes a lot of sense. That may have been why the dash was not included in the regex to begin with, since it requires escaping in the create command. I don't usually recommend using a dash in a database, and some systems convert them to underscores.

I'll go ahead and merge this. Thanks!